enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -212,6 +242,76 @@ These changes have already been implemented, and the initial PR for them can be
+https://github.com/openshift/kubernetes/blob/3c62f738ce74a624d46b4f73f25d6c15b3a80a2b/openshift-kube-apiserver/openshiftkubeapiserver/patch.go#L71
+
+In order to prevent misleading logs about informers that failed to start or failure to connect to the oauth-apiserver, the following changes to this patch are to be made:
+
+- Informers for the `Group` API are only configured and started as part of the first run of the `authorization.openshift.io/RestrictSubjectBindings` admission plugin validation loop. This makes it such that the informer will not be configured or attempt to start when the admission plugin is disabled.
+- The post-start hook that checks for oauth-apiserver connectivity will be skipped if the `Authentication` resource `.spec.type` is set to `OIDC`. This will prevent logs in the kube-apiserver associated with not being able to connect to the oauth-apiserver, which we know should not be running when OIDC is enabled.
+
+##### Changes to the cluster-kube-apiserver-operator
+
+When authentication type is set to OIDC, the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins will be disabled.
+
+This will be done through updates to the appropriate config observers to update the `KubeAPIServerConfig.apiServerArguments` map to:
+
+- Remove the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins from the `--enable-admission-plugins` argument
+- Add the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins to the `--disable-admission-plugins` argument
+
+##### Changes to the cluster-authentication-operator
+
+A couple changes will need to be made to the cluster-authentication-operator. Mainly:
+- Ownership/Management of the `rolebindingrestrictions.authorization.openshift.io` CustomResourceDefinition
+- Blocking OIDC configuration due to existence of `RoleBindingRestriction` resources and communicating it via the `Authentication` resources new `OIDCConfig` status field
+
+To support the need to remove the `rolebindingrestrictions.authorization.openshift.io` CustomResourceDefinition when configuring the authentication type as OIDC, the cluster-authentication-operator will be updated to manage this CRD.
+
+This will mean vendoring the generated CRD manifests as outlined in https://github.com/openshift/api/tree/master?tab=readme-ov-file#vendoring-generated-manifests-into-other-repositories and adding a new controller to manage the CRD.
+
+Managing the CRD will consist of ensuring that the CRD is present on the cluster, and matches the desired manifest, when the authentication type is _not_ OIDC, and ensuring the CRD is not present present on the cluster when the authentication type _is_ OIDC and OIDC configuration has been successfully rolled out. 
```

Comment:
thinking through cases, is it actually
1. internal oath server is desired, ensure crd/rolebindingrestriction
2. internal oauth server is configured and not desired and any rolebindingrestrictions currently exist, ensure crd/rolebindingrestriction.  This supports a migration case to avoid removing groups and users while the tokens are still honored
3. internal oauth server is configured and not desired and no rolebindingrestrictions currently exist, remove crd/rolebindingrestriction.  This supports a migration case.
4. internal oauth server is not configuration and not desired, remove crd/rolebindingrestriction

Reply:
Yeah, that all sounds right. I'll update this section to be more clear on the different scenarios.

Reply:
Updated in [c32abf3](https://github.com/openshift/enhancements/pull/1726/commits/c32abf3d79d816cb71fcfd653c977974779e5a71)